hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetworkTopologyImpl.java

```diff
@@ -556,17 +571,17 @@ private Node chooseNodeInternal(String scope, int leafIndex,
     int nodeIndex;
     if (leafIndex >= 0) {
       nodeIndex = leafIndex % availableNodes;
-      ret = ((InnerNode)scopeNode).getLeaf(nodeIndex, excludedScope,
+      ret = ((InnerNode)scopeNode).getLeaf(nodeIndex, mutableExcludedScopes,
           mutableExNodes, ancestorGen);
     } else {
       nodeIndex = ThreadLocalRandom.current().nextInt(availableNodes);
-      ret = ((InnerNode)scopeNode).getLeaf(nodeIndex, excludedScope,
+      ret = ((InnerNode)scopeNode).getLeaf(nodeIndex, mutableExcludedScopes,
           mutableExNodes, ancestorGen);
     }
     LOG.debug("Choosing node[index={},random={}] from \"{}\" available nodes" +
             " scope=\"{}\", excludedScope=\"{}\", excludeNodes=\"{}\".",
```

Comment:
same as above.

Reply:
The value of list is printed. 

2019-09-10 19:16:53,044 INFO  net.NetworkTopology (NetworkTopologyImpl.java:chooseNodeInternal(561)) - Choosing node[index=0,random=true] from "4" available nodes scope="/", excludedScope="[/rack0/node4]", excludeNodes="[c96d3d66-259b-4eae-9a19-587aec77f5d6{ip: 253.157.233.249, host: node10, networkLocation: /rack2, certSerialId: null}, 38f20ed2-c2f4-4665-a157-492aa74a80dc{ip: 20.53.242.116, host: node11, networkLocation: /rack2, certSerialId: null}, 7de18ff0-154d-4861-b682-e0780f60679c{ip: 210.105.89.96, host: node7, networkLocation: /rack1, certSerialId: null}]".